web/src/muted_topics_ui.js

```diff
@@ -16,7 +17,11 @@ export function rerender_for_muted_topic(old_muted_topics) {
     if (message_lists.current !== message_lists.home) {
         message_lists.home.update_muting_and_rerender();
     }
-    if (overlays.settings_open() && settings_muted_topics.loaded) {
+    if (
+        !page_params.development_environment &&
```

Comment:
Do we need this development_environment check? It seems out of place.

Reply:
The other `development_environment` guards are probably correct, but I think my read of https://chat.zulip.org/#narrow/stream/101-design/topic/unmuting.20in.20muted.20streams/near/1553245 is that we can probably just plan to merge this early next week, and it is probably simplest to just remove them.

Reply:
> it is probably simplest to just remove them.

I have removed the `development_environment` guards in favour of the plan you mentioned.

> Do we need this development_environment check? It seems out of place.

Earlier, It was needed **temporarily** because of the different behaviour in dev and non-dev environment.
As [mentioned by alya](https://github.com/zulip/zulip/pull/25127#issuecomment-1513760715) we needed to implement the following two behaviours in dev env:
1. Topics doesn't get resorted when you change the status in this UI.
2. When we set a topic to "Default for stream", it stays in the UI until we refresh or navigate away.

It wasn't required in non-dev environment.

In short:
Finally I have removed that code block because the list should be re-populated only if we refresh or navgate away.